src/modules/SourceManager.rb

```diff
@@ -272,7 +272,7 @@ def createSource(url)
 
       UI.CloseDialog if !Mode.commandline
 
-      if Builtins.size(@newSources).zero?
+      if @newSources.nil? || @newSources == -1 || @newSources.size.zero?
```

Comment:
So `@newSources` can be nil, or -1, or a string... that's pretty weird already, so I would like to see more defensive code here (and a comment). This could still explode if `@newSources` would be, for example, -2. And looking at the history, that wouldn't surprise me.

Reply:
well, I check pkg bindings about possible values and also plan to document it. I think in general this pkg-bindings API sucks, as it looks like in past it was list and empty list mean problem with source, but it was changed. So I plan second PR that will make it clear in Pkg method documentation.

Reply:
And that's exactly why I don't trust the result of that call to `SourceScan` to stay consistent in the future. Of course, I trust you checked and the only possible values RIGHT NOW are those. But that changed already in the past and could change in the future.

So I would like to see a robust condition checking the result has the exact form we expect, raising the pop-up if not. Something like
`if !@newSources.kind_of?(::String) || @newSources.size.zero?`
or maybe
`if !@newSources.respond_to?(:size) || @newSources.size.zero?`

Just in case tomorrow somebody decides that some kind of errors should be reported as -2.

Reply:
ok, in fact it expects List. See linked pr where I try to clarify it.